contracts/truefi2/LoanFactory2.sol

```diff
@@ -198,7 +198,7 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
         ITrueFiPool2 _pool,
         address _borrower,
         uint256 _debt
-    ) external override onlyTCA {
+    ) external override onlyTCA returns (address) {
```

Comment:
This function should return an IDebtToken to help with compiler type-checking. Otherwise all the places we want to call this would need to cast it manually. That would be more error-prone than casting once in the factory create function.

Reply:
This probably compiles right now, but it will fail to run because you're calling createDebtToken() from LoanToken instead of TCA.

A general solution discussed in yesterday's meeting would be to delegate these checks to BorrowingMutex:
- TrueFiPool2's onlyLenderOrFTLAgencyOrTrueCreditAgency
- here in LoanFactory2's onlyTCA and onlyFTLA
- LoanToken2's onlyLenderOrFTLAgency

BorrowingMutex has two mappings that I think are perfect for gating which contracts can interact with pools and borrowers:
- `canLock(msg.sender)` -> onlyCanLock guards which agencies can initiate lending from supported pools
- `msg.sender == locker(borrower)` -> onlyLocker guards existing loans from being interfered with by unrelated callers (e.g., other agencies or malicious actors).

I think we should handle these conversions in a separate PR and check it in before continuing with this PR. Sorry I didn't notice this `onlyTCA` dependency while splitting tasks.

Do you think this PR sequencing makes sense?

Reply:
Actually let's check this PR in with something like `onlyTCAOrLoan` using a simple manual guard for now, and we will move to the full borrowing mutex guard later.